scripts/pipeline/orthology_benchmark.py

```diff
@@ -175,6 +177,64 @@ def get_core_names(species_names: List[str], host: str, port: int, user: str) ->
     return core_names
 
 
+def get_gtf_file(core_name: str, source_dir: str, target_dir: str) -> None:
+    """Finds and copies a GTF file for a specified core db.
+
+    Note:
+        The code utilises the structure of `production/ensemblftp` or MC's personal dir
+        to speed up file search.
+
+    Args:
+        core_name: Core db name.
+        source_dir: Path to the directory containing subdirectories with GTF files.
+            `production/ensemblftp` or MC's `nobackup/release_dumps`.
+        target_dir: Path to the directory where the GTF file will be copied.
+
+    Raises:
+        RuntimeError: If command to copy a file fails for any reason.
+
+    Warns:
+        UserWarning: If the GTF file was not found.
+
+    """
+    species_name = core_name.split("_core_")[0]
+    release = core_name.split("_core_")[1].split("_")[0]
+
+    parent_dir = os.path.join(source_dir, "release-" + release)
+    # gtf_dirs contains all subdirs which could contain the gtf file
+    # Vertebrates in `production/ensemblftp`:
+    gtf_dirs = [os.path.join(parent_dir, "gtf", species_name)]
+    # Vertebrates in MC's personal dir and
+    # all other divisions whether in `production/ensemblftp` or MC's dir
+    potential_divisions = [f.name for f in os.scandir(parent_dir) if f.is_dir()]
+    for division in potential_divisions:
+        gtf_dirs.append(os.path.join(parent_dir, division, "gtf", species_name))
+
+    for directory in gtf_dirs:
+        try:
+            gtf_file = [file for file in glob.glob(os.path.join(directory, "*"))
+                        if file.endswith(f".{release}.gtf.gz")][0]
+        except IndexError:
+            continue
+
+        try:
+            cmd = f"cp {gtf_file} {target_dir}"
```

Comment:
If `target_dir` doesn't exist, `subprocess.run` fails and `RuntimeError` is raised. Would you strongly advise going with `shutil.copy` or `shutil.copyfile` instead?

Reply:
I would indeed advise to use an `shutil` function such as `shutil.copy` or `shutil.copyfile`. In this case `shutil.copy` would be more similar to the existing command since it allows the target to be a directory.

One caveat: if the target directory doesn't exist but its parent does, the copied file will be saved to a file with the same name as the intended target directory. To avoid this, you could ensure the target directory exists beforehand with `os.makedirs` as follows:
```
os.makedirs(target_dir, exist_ok=True)
shutil.copy(gtf_file, target_dir)
```
This doesn't handle the check of the `result` in the same way that the existing code does, but I'd advise using an `shutil` copy function anyway because it will make the code more concise.

Reply:
It probably doesn't handle errors in the same way as `subprocess` does, either. So this requires a bit more rewriting. Just wanted to check if it was worth doing it. Thank you for your extensive reply, @twalsh-ebi.